### DISCUSSION TITLE
Why does GitHub not recognize my dependabot.yml file?

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#28261] [New to GitHub] [answered]
A few hours ago did I create a `dependabot.yml` file within the `.github` directory of my repository to enable Dependabot updates.

However, checking the Dependabot Tab in the dependency graph section shows me the same button that is displayed whenever you enabled Dependabot, but didn't create a Dependabot.yml file yet.

![image](https://user-images.githubusercontent.com/11576465/182721841-eb059832-b783-400d-b553-c812e82d4f1b.png)

Clicking it will simply open the same prompt for when you create a new file. Out of safety did I not try to save that file to see if it gets oevrriden.

The dependabot.yml is present within the repository and is in fact on the default branch (in my case master). The format also seems to be valid without typos or similar.

You can find it here: https://github.com/Andre601/AdvancedServerList/blob/master/.github/dependabot.yml

I'm seriously confused why this happens. It never did that before and I can't find the cause of those problems...

**Comment by kurnakovv** [comment, marked as answer]
@Andre601 Try to copy the code, then delete this file, and then create a file in github (and paste your code here), it helped me once

**Reply by Andre601** [reply]
I think there is generally something wrong with the dependabot.yml file creation.
Because when I create it through the button is there no code highlighting for the YAML file, so GitHub's editor isn't recognizing it as a valid YAML file.

Manual creation through the `Add File -> Create new file` button is showing the code highlighting.
This manual creation also fixes the dependabot file not being recognized.

So I guess something in that creation process is messed up.

**Reply by kurnakovv** [reply]
Yea, it's strange, but I hope later it will fix.
Thank you for "Marked as answer" and have a nice day ;)

**Reply by queenofcorgis** [reply]
Nice work @KurnakovMaksim - way to get the solve!

**Comment by distantnative** [comment]
We’re facing the same problem here still: https://github.com/getkirby/kirby

Recreated the config file via the UI, also directly on the main branch. But all no success. Any ideas?

**Reply by kurnakovv** [reply]
@distantnative dependabot work, but you have limit for pull requests:
`open-pull-requests-limit: 10`
you already have 10 pull requests on the project, you can just increase the limit, for example:
`open-pull-requests-limit: 50`

**Reply by distantnative** [reply]
Oh! We totally thought this relates to how many Dependabot PRs not PRs in general. Will try that then. 👍

**Reply by distantnative** [reply]
Hmm, according to the config docs the limit indeed doesn't refer to all PRs but just those created by Dependabot.

However, I updated the config, but GitHub still does not recognise the file as shown by this UI:
<img width="1250" alt="Screenshot 2022-08-31 at 08 31 12" src="https://user-images.githubusercontent.com/3788865/187609479-9f7a7a79-a546-47ea-b5af-0b13906037a8.png">

**Reply by kurnakovv** [reply]
Hmm, strange, try to recreate this file again, for experience

**Reply by kurnakovv** [reply]
btw, why do you need this config?
`target-branch: "develop"`
mb main branch is not "develop"?

**Reply by distantnative** [reply]
@KurnakovMaksim It's the third time we've (re-)created it 😄

Dependabot updates should go into our regular releases. That's why they need to be merged into develop first which then lands on main at every release.

**Reply by kurnakovv** [reply]
I understand. You need update dependabot in develop branch, after that should work https://github.com/getkirby/kirby/blob/develop/.github/dependabot.yml#L17
https://github.com/getkirby/kirby/blob/develop/.github/dependabot.yml#L28

**Reply by distantnative** [reply]
I did, but I am afraid it still doesn't pick up the config file.

**Reply by kurnakovv** [reply]
strange, because dependabot updated: https://github.com/getkirby/kirby/runs/8120627101
mb you haven't old dependencies? :D

**Reply by distantnative** [reply]
I hoped that was the case, but still no dependabot activities while we are on `develop` branch on vite 3.0.9 https://github.com/getkirby/kirby/blob/develop/panel/package.json#L53 with 3.1.0 having been released 5 days ago already https://github.com/vitejs/vite/releases/tag/v3.1.0
